coveralls/api.py

```diff
@@ -202,6 +204,21 @@ def wear(self, dry_run=False):
         verify = not bool(os.environ.get('COVERALLS_SKIP_SSL_VERIFY'))
         response = requests.post(endpoint, files={'json_file': json_string},
                                  verify=verify)
+
+        # check and adjust/resubmit if submission looks like it
+        # failed due to resubmission (non-unique)
+        if response.status_code == 422:
+            self.config['service_job_id'] = '{}-{}'.format(
+                self.config['service_job_id'], random.randint(0, sys.maxsize))
```

Comment:
Wouldn't this lead to a potential `KeyError` on GitHub Actions?
Because the `service_job_id` is only set if `job` is set in `Coveralls.__init__()`:
https://github.com/coveralls-clients/coveralls-python/blob/2.2.0/coveralls/api.py#L56-L57
However `job` is always `None` in `Coveralls.load_config_from_github()`:
https://github.com/coveralls-clients/coveralls-python/blob/2.2.0/coveralls/api.py#L92-L109

Reply:
You may be right - looks like github actions are one of/the only ones that requires use of a token, and then submits this using the second method documented at https://docs.coveralls.io/api-introduction. Have you seen this occur in practice?

Reply:
Yes I think I saw it happening with when using `service_name=github` and getting a 422.
The fix for the `KeyError` was to simply provide `service_job_id` a value (`None` in my case) to the `Coveralls` constructor. That was enough for me. See below for the fix including regression tests.
https://github.com/AndreMiras/coveralls-python-action/commit/50af55ed0bf1537fb4404f4ccc550befd17156bb
Since it could well be me misusing the package, I guess we can simply acknowledge this bug/feature and take no actions until someone else complains